src/components/App.tsx

```diff
@@ -1,16 +1,19 @@
 import React from "react";
 import "../App.css";
 import Header from "./general/header";
-import Compare from "./company/compare";
-import { Card, CardContent, Container } from "@mui/material";
+import { Container, Card, CardContent } from "@mui/material";
+import { Outlet } from "react-router-dom";
+import { BackendApi, IBackendAPI } from "../services/api";
 
 function App() {
+  const [api] = React.useState<IBackendAPI>(BackendApi);
+
   return (
     <Container maxWidth="xl" className="app text-centered">
       <Header />
       <Card sx={{ mt: 2, pt: 2 }}>
         <CardContent>
-          <Compare />
+          <Outlet context={{ api }} />
```

Comment:
Hier voeg ik een outlet toe. De outlet zorgt ervoor dat het component van de route ingeladen wordt.
De context werkt zoals een gewone context in React, maar zit ingebouwd in React router.
De API wordt hier meegegeven in de context zodat die kan worden aangesproken.